packages/compass-collection/src/index.ts

```diff
@@ -1,36 +1,39 @@
-import type AppRegistry from 'hadron-app-registry';
-import CollectionTabsPlugin from './plugin';
-import CollectionTabsStore from './stores/tabs';
 import CollectionTab from './components/collection-tab';
-import { configureStore } from './stores/collection-tab';
+import { activatePlugin as activateCollectionTabPlugin } from './stores/collection-tab';
+import { registerHadronPlugin } from 'hadron-app-registry';
+import { dataServiceLocator } from 'mongodb-data-service/provider';
+import { mongoDBInstanceLocator } from '@mongodb-js/compass-app-stores/provider';
+import type { DataService } from 'mongodb-data-service';
 
-const COLLECTION_TAB_ROLE = {
-  name: 'CollectionTab',
-  component: CollectionTab,
-  configureStore,
-};
+export const CollectionTabPlugin = registerHadronPlugin(
+  {
+    name: 'CollectionTab',
+    component: CollectionTab,
+    activate: activateCollectionTabPlugin,
+  },
+  {
+    dataService: dataServiceLocator as typeof dataServiceLocator<
```

Comment:
```suggestion
    dataService: dataServiceLocator as DataServiceLocator<
```

(depcheck :))

Reply:
Merged with main and updated that, thanks again for highlighting!